src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -499,15 +499,7 @@ private void BufferHeaders(HttpRequestMessage request)
             _sendBuffer.Commit(2);
 
             HttpMethod normalizedMethod = HttpMethod.Normalize(request.Method);
-            if (normalizedMethod.Http3EncodedBytes != null)
-            {
-                BufferBytes(normalizedMethod.Http3EncodedBytes);
-            }
-            else
-            {
-                BufferLiteralHeaderWithStaticNameReference(H3StaticTable.MethodGet, normalizedMethod.Method);
```

Comment:
@scalablecory, is this an ok change?

Reply:
It's potentially less efficient if a customer has already inefficient code and isn't caching custom `HttpMethod` instances. Previously this would never allocate for a custom `HttpMethod`, and now it will.

I'm fine with this change. For the common case it removes a branch, which is probably a worthwhile tradeoff.